packages/cdktf/lib/tokens/private/resolve.ts

```diff
@@ -21,6 +21,32 @@ import {
 import { TokenMap } from "./token-map";
 import { Token } from "../token";
 
+const LIST_ERROR_EXPLANATION = `In CDKTF we represent lists where the value is only known at runtime (versus compile / synth time) as
+Arrays with a single element that is a string token, e.g. ["Token.1"]. This is because at compile time we
+don't know the length of the list, so far CDKTF did not invoke Terraform to communicate with the cloud provider.
+This is done at a later stage on the synthesized static JSON file.
+As we don't know the length of the list not the content we can not differenciate if the list was accessed at the first index,
+the last index, or as part of a loop. To avoid this ambiguity:
+
+- If you want to access a singular item use 'Fn.element(list, 0)' (not 'list[0]')
+- If you want to loop over the list use 'TerraformIterator.fromList(list)' (not 'for (const item of list)' or 'list.forEach(item => ...)')
+
+To learn more about tokens see https://developer.hashicorp.com/terraform/cdktf/concepts/tokens
+To learn more about iterators see https://developer.hashicorp.com/terraform/cdktf/concepts/iterators`;
+
+const MAP_ERROR_EXPLANATION = `In CDKTF we represent maps where the value is only known at runtime (versus compile / synth time) as
+Objects with a single key-value pair where the value is a string token, e.g. { "&{TfToken[Token.1]}": "String Map Token Value" }. This is because at compile time we
```

Comment:
```suggestion
pbjects with a single key-value pair where the value is a string token, e.g. { "&{TfToken[Token.1]}": "String Map Token Value" }. This is because at compile time we
```

also: "where the value is a string token" + your example has the token in the key – Is it the key or the value that holds the token?